server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala

```diff
@@ -606,4 +607,66 @@ trait MailboxSetMethodContract {
          |    "c1"]]
          |}""".stripMargin)
   }
+
+  @Test
+  def mailboxSetShouldNotCreateMailboxWhenMailboxItShared(server: GuiceJamesServer): Unit = {
```

Comment:
Name of test is wrong...

How about `mailboxSetShouldNotCreateChildMailboxWhenSharedParentMailbox` ? Not sure it's the best though, opened to better names